src/sparseml/exporters/kv_cache_injector.py

```diff
@@ -80,63 +89,53 @@ def __init__(
         :param model_path: The path to the directory containing the model.
         :param inplace: If True, the model will be modified in place.
             If False, a copy of the model will be made and modified.
+        :param kwargs: (Optionally) the parameters for the KV cache injection
+            if no `model_path` is provided.
         """
         self.inplace = inplace
-        if model_path:
-            self.config = self.get_config(model_path)
-            if not self.config["model_type"] in _SUPPORTED_ARCHITECTURES:
-                raise ValueError(
-                    f"Model type {self.config.model_type} is currently not supported. "
-                    f"Supported model types: {_SUPPORTED_ARCHITECTURES}."
-                    f"Proceeding with transformation, but may require additional "
-                    f"customization..."
-                )
-
-            num_attention_heads = self.config["num_attention_heads"]
-            hidden_size_kv_cache = self.config["hidden_size"] // num_attention_heads
+
+        if model_path is not None:
+            # get the parameters from the config
+            self.config = get_kv_cache_config(model_path)
```

Comment:
If I understand correctly, the `kv_cache_config` is a hardcoded config for each model type. the model_path here is just used to determined the model type right?

Reply:
Yes, exactly. `model_path` is a path to the directory that contains HF's `config.json`. Inside this config, you will find information about the model type and the architecture (num of attention heads etc). All this information is being used to pick the correct "hardcoded config" (semi-hardcoded, since if you have the same model of different size, the "hardcoded config" will be robust against it) and then pass it on in the injection process.